pkgs/by-name/fi/filius/package.nix

```diff
@@ -0,0 +1,115 @@
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "filius";
+
+  # UPDATE instructions
+  #
+  # - Open https://www.lernsoftware-filius.de/Herunterladen and copy download link to clipboard.
+  #   (e.g. https://www.lernsoftware-filius.de/downloads/Setup/filius-2.5.1.zip)
+  # - identify the new version string.
+  version = "2.5.1";
+
+  # - Open http://web.archive.org and paste download link from clipboard into "Save Page Now" field and hit the "Save Page" button.
+  # - Unselect "Save Error Pages" and hit "Save Page" again.
+  # - Wait for the archive link to be generated and copy it to the url field - adjust hash accordingly.
+  src = fetchzip {
+    url = "https://web.archive.org/web/20240623122638/https://www.lernsoftware-filius.de/downloads/Setup/filius-2.5.1.zip";
+    hash = "sha256-UrtmL0jHf86vXXCtRJoaXAyElBvUUF1jGpBcO0GV6Lg=";
+    stripRoot = false;
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Filius";
+      exec = "filius";
+      icon = "filius";
+      desktopName = "Filius";
+      comment = "Computer Network Simulator";
+      categories = [ "Education" ];
+    })
+  ];
+
+  dontConfigure = true;
+  dontBuild = true;
+  # Tests need a DISPLAY
+  doCheck = false;
+
+  installPhase = ''
+    runHook preInstall
+
+    export CUSTOM_LIBS=$out/share/java
+    export JAR=$CUSTOM_LIBS/filius.jar
+
+    # does create also folders
+    install -D filius.jar $CUSTOM_LIBS/filius.sh
+
+    cp -r * $CUSTOM_LIBS/
+    cp -r *.* $CUSTOM_LIBS/
+
+    makeWrapper ${jre}/bin/java $out/bin/filius \
+      --add-flags "-Duser.dir=$CUSTOM_LIBS/" \
+      --add-flags "-Xmx512M" \
+      --add-flags "-jar $JAR" \
+      --add-flags '-n -wd $HOME' \
+      --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=lcd'
+
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    install -Dm444 ${icon128} $out/share/icons/hicolor/128x128/apps/filius.png
+    install -Dm444 ${icon16} $out/share/icons/hicolor/16x16/apps/filius.ico
+    install -Dm444 ${icon32} $out/share/icons/hicolor/32x32/apps/filius.ico
+  '';
+
+  meta = {
+    homepage = "https://www.lernsoftware-filius.de";
+    downloadPage = "https://www.lernsoftware-filius.de/Herunterladen";
+    description = "A Computer Network Simulator for Secondary Schools";
+    longDescription = ''
+      With the software tool Filius, you can design computer networks yourself,
+      simulate the exchange of messages in them and thus explore their structure
+      and functionality experimentally. The target group are learners at secondary
+      level in general education schools. Filius enables learning activities that
+      are designed to support discovery-based learning in particular.";
+    '';
+    license = with lib.licenses; [
+      gpl2
+      gpl3
+    ];
+    maintainers = with lib.maintainers; [ rcmlz ];
+    platforms = lib.platforms.all;
```

Comment:
i guess it doesn't run all all platforms (?) but the icons only work on systems supporting freedesktop